Source/cmake/FindGIDocgen.cmake

```diff
@@ -50,6 +50,12 @@ local copy without needing to vendor it in the project source tree.
 
 #]=======================================================================]
 
+# Force CMP0116: cmake will automatically translate absolute paths in
+# depfiles to paths relative to the sub-build directory. Ninja expects
+# relative paths in depfiles.
+# This behavior is already the default when targeting cmake 3.20+.
+cmake_policy(SET CMP0116 NEW)
+
```

Comment:
Instead of adding this here, I think it's better to change from `OLD` to `NEW` in the top-level `CMakeLists.txt` file. It was added in 6cd89696b5d406c1a3d9a7a9bbb18fda9284fa1f ([bug #240535](https://bugs.webkit.org/show_bug.cgi?id=240535)) to try use dependency files with `glib-compile-resources` but that didn't work as well as expected. After your changes in [26786](https://github.com/WebKit/WebKit/pull/26786) we no longer will want anyway to set the policy to `OLD` and it will be better to always use `NEW` for CMake <3.20.

Reply:
Actually, feel free to do the suggested change as part of https://github.com/WebKit/WebKit/pull/26786 if you think that's easier.

Reply:
Do you mean in the CMakeLists.txt at the root of WebKit? I would be in favor of upping its required version... afterall, bmalloc/libpas is already requiring a higher version than the root CMakeLists.txt. I have gone with this approach only to play incrementally, as that way the policy is scoped only to that file. (Yes, include()'d files have a policy scope, even though functions don't. CMake is janky like that.)

Reply:
Our bots on Windows are 3.29.0 and on PlayStation 3.27.1 so we would not hit an issue.

Reply:
I later realized that we're still stuck on 3.16 since that's what's available in Debian 11 which we still have to support until June.

Also I realized that a global fix was attempted previously, but had a mistake: https://github.com/WebKit/WebKit/commit/6cd89696b5d406c1a3d9a7a9bbb18fda9284fa1f#diff-1e7de1ae2d059d21e1dd75d5812d5a34b0222cef273b7c3a2af62eb747f9d20aR13-R18 The policy was set to OLD, which has no effect. It should be set to NEW.

Then I also noticed that setting it to NEW broke my other PR (https://github.com/WebKit/WebKit/pull/26786). You've gotta love build systems. But removing the code that makes the relative paths and letting CMake compute them fixes it back, so that's an easy fix. At this point it might make more sense to merge both patches into one.

And on a general more positive note, on my computer, with those two fixes and a third one I just figure out I'm now re-building without changes under a second, so there is promise.

Reply:
We can already update the minimum CMake version requirement in `main`, because it is *stable* releases the ones that have to be buildable in the supported Linux distribution versions. The next stable branch (2.46) will be created sometime in August-September 2024, so that's already after support for Debian 11 would be dropped.

Also, I agree on merging both patches into one 👌🏼